### DISCUSSION TITLE
Disable the Double click

### DISCUSSION THREAD

**Discussion by jgVitc** [stephanrauh/ngx-extended-pdf-viewer#1398] [Q&A]
Hi,
Is there a way to disable the double click so that it doesn't zoom in or out?

**Comment by stephanrauh** [comment]
I suppose the best solution is adding an event listener on `#viewer`. Something like this untested pseudo code:

```typescript
document.querySelector('#viewer').addEventListener('dblclick', 
                                       (evt) => {evt.preventDefault(); evt.stopPropagation();}, true);
```

Maybe you also need to use the parent element of `#viewer`, i.e. `#viewerContainer`. In any case, @mahmooudali7 reports in #970 that this approach does the trick.

By the way, please apologize the sluggish answer: A nasty flu sent me to bed this week.

**Reply by TLucasIND** [reply]
Hi Stephan, I tried the above solution. The doubleclick gets caught correctly and I tested evt.preventDefault();
`evt.defaultPrevented();` returns true. However the zoom still happens. I call the addEventListener in the ngOnInit()-Method. 
Do you have a hint for me?

**Reply by stephanrauh** [reply]
I'd try to use `ngAfterViewInit()`, and if that doesn't work, I'd try to go one div layer up. If I'm not mistaken, but click event bubbles from the upmost (and outermost) layer to the bottom, so stopping it at a higher layer should prevent it to trickle down.

**Reply by TLucasIND** [reply]
After some time I returned to the problem. I went all the div layers up and still had no luck.
What might be the problem? I tried to put the registration of the callback in the two different initialization-phases ngOnInit() and ngAfterViewInit.
I also tried  `evt.stopImmediatePropagation();`. Still no luck :( 
However the event is caught properly, so the evt.-calls are the problem.

Greets, Timo

**Reply by TLucasIND** [reply]
Awesome, I found the bug. The last boolean argument was not in the code above. It's mandatory here to .

```
document.addEventListener('dblclick', (evt) => {
      evt.preventDefault(); 
      evt.stopPropagation();
    }, true);
```

> The options argument sets listener-specific options. For compatibility this can be a boolean, in which case the method behaves exactly as if the value was specified as options's capture.

> When set to true, options's capture prevents callback from being invoked when the event's eventPhase attribute value is BUBBLING_PHASE. When false (or not present), callback will not be invoked when event's eventPhase attribute value is CAPTURING_PHASE. Either way, callback will be invoked if event's eventPhase attribute value is AT_TARGET.

**Reply by stephanrauh** [reply]
I'm glad you've found and solved the problem!

Is there anything you want me to do? Like updating my answer, improving the documentation, or implementing the feature in the library? I know your problem is a thing of the past, but maybe I can do something to help other developers.

**Reply by TLucasIND** [reply]
Maybe it would be nice to have the option to toggle the double-click-zoom. But I'm not sure, if you use the doubleclick additionally in  any other way. Would it be a big expense for you to add it?

**Reply by stephanrauh** [reply]
Adding a feature toggle is not a big deal. I suppose the most difficult part of it is finding an intuitive name for the attribute. Actually, I implemented the double-click zoom for mobile devices. I always enjoyed double-tapping my iPhone to enlarge the text to the optimal size. If I'm not mistaken, they've removed this feature recently. That's a pity, but it also means users stop expecting the double-tap zoom.
